### ISSUE TITLE
Incompatibility issues with Samplers / Schedulers in AUTO1111 Version 1.9+

### ISSUE THREAD

**Issue by Gecktendo** [Kilvoctu/aiyabot#264]
**Describe the bug**  
Automatic 1.9 recently rolled out a change to their Samplers where they separate the scheduler type with the sampler itself so you can pick two separately. This is causing issues with the bot since the bot still bundles sampler and scheduler types together.

When you try to generate images in 1.9, you get an error message saying there's a problem with the "images" parameter.

---

**Environment**
 Windows 11

---

**Console logs**

**Comment by Kilvoctu** [comment]
Heya I updated my webui and didn't encounter this issue
Screenshot of my webui and aiya log:
![image](https://github.com/Kilvoctu/aiyabot/assets/2993060/f0c21b73-06bc-4d07-b0a5-c577abfc2328)

What settings do I need to apply to recreate the problem?

**Comment by Kaiybo** [comment]
Same problem here.  Getting a 
txt2img failed
An invalid parameter was found!
'images'
failure since I had the Webui updated

as for the settings that is causing this, I'm not entirely sure. I'm still looking for a workaround.

**Comment by Kaiybo** [comment]
Found a workaround. Ok so for a discord channel where I didn't have a sampler set by default, it would fail. after I did a /settings and set the sampler to DPM++ 2M, it began working again.

**Comment by Gecktendo** [comment]
Here's what I get when I try to set the sampler.  Nothing shows up.
![Bug](https://github.com/Kilvoctu/aiyabot/assets/125915576/dd88ae54-0e57-462e-bf9a-ef668d528119)

**Comment by Gecktendo** [comment]
Here's what I get when I try to choose the sampler.

However, when you manually type the sampler in, it works.
![Bug](https://github.com/Kilvoctu/aiyabot/assets/125915576/2566aa4b-e6cc-4798-b1e4-bd46a414efab)

**Comment by Gecktendo** [comment]
> Found a workaround. Ok so for a discord channel where I didn't have a sampler set by default, it would fail. after I did a /settings and set the sampler to DPM++ 2M, it began working again.

Yeah this was the case for me too. Seems like that's what fixes it.

**Comment by wizz13150** [comment]
> Found a workaround. Ok so for a discord channel where I didn't have a sampler set by default, it would fail. after I did a /settings and set the sampler to DPM++ 2M, it began working again.

Hey,

indeed "DPM++ 2M" is a correct sampler name for a1111 v1.9, so it's working.
But the scheduler is not mentionned, so the webui takes the "Automatic" choice by default.

We need to add a 'scheduler' option for /Draw.
It's not that hard.

Buuut, if we want to have the 'scheduler' right after the sampler in the Edit button, then we need to increment all the tuple numbers. This is the boring part.
I'll do it for my fork and post the commit here. _But_ my fork is now too far from the original bot, the lines won't match and will need to be adapted to the current version of the bot.

Cheers ! 🥂

**Comment by sean-ellis** [comment]
Sorry for bumping an old post, but any chance we could get schedulers still?
